vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/PriorityByWorkflow.java

```diff
@@ -0,0 +1,140 @@
+package ca.on.oicr.gsi.vidarr.server;
+
+import static java.util.Map.Entry.comparingByValue;
+
+import ca.on.oicr.gsi.Pair;
+import ca.on.oicr.gsi.vidarr.BasicType;
+import ca.on.oicr.gsi.vidarr.ConsumableResource;
+import ca.on.oicr.gsi.vidarr.ConsumableResourceResponse;
+import com.fasterxml.jackson.databind.JsonNode;
+import io.prometheus.client.Gauge;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.SortedSet;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentSkipListSet;
+
+
+final class PriorityByWorkflow implements ConsumableResource {
+
+  private List acceptedPriorities = Arrays.asList(1, 2, 3, 4);
+
+  private static final class WaitingState {
+
+    private final SortedSet<SimpleEntry<String, Integer>> waiting = new ConcurrentSkipListSet<SimpleEntry<String, Integer>>(comparingByValue()){
+
+      @Override
+      public boolean add(SimpleEntry<String, Integer> simpleEntry) {
+        super.remove(this.getByKey(simpleEntry.getKey()));
+        return super.add(simpleEntry);
+      }
+
+      @Override
+      public boolean remove(Object o) {
+        try {
+          String oEntry = (String) o;
+          for (SimpleEntry entry : this){
+            if (entry.getKey().equals(oEntry)) {super.remove(entry);}
+          }
+        } catch (Exception e){
+          return false;
+        }
+        return true;
+      }
+
+      public SimpleEntry getByKey(String simpleEntryKey){
+        for (SimpleEntry entry : this){
+          if (entry.getKey().equals(simpleEntryKey)) {return(entry);}
+        }
+        return null;
+      }
+    };
+  }
+
+
+  //remove?
+  private static final Gauge currentInPriorityWaitingCount =
+      Gauge.build(
+              "vidarr_in_priority_waiting_per_workflow_current",
+              "The current number of workflows that are on priority waiting to run.")
+          .labelNames("workflow")
+          .register();
+
+  private final Map<String, WaitingState> workflows = new ConcurrentHashMap<>();
+
+  @Override
+  public Optional<Pair<String, BasicType>> inputFromSubmitter() {
+    return Optional.of(new Pair<String, BasicType>("Priority", BasicType.INTEGER));
+  }
+
+  @Override
+  public void recover(String workflowName, String workflowVersion, String vidarrId, Optional<JsonNode> resourceJson) {
+    final var stateWaiting = workflows.computeIfAbsent(workflowName, k -> new WaitingState()).waiting;
+    // since we just created it if it doesn't exist, no need for null check here
+
+    int workflowPriority;
+
+    if (resourceJson.isPresent()) {
+      JsonNode nodeInput = resourceJson.get();
+      workflowPriority = nodeInput.get("Priority").asInt();
```

Comment:
I'm getting an error here when I've submitted a workflow run with `"consumableResources": { "Priority": 2 }` , saying that it can't get key `Priority` because the value of `nodeInput` is `2`. Looking at the [ConsumableResourceChecker here](https://github.com/oicr-gsi/vidarr/blob/master/vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/ConsumableResourceChecker.java#L83), it makes sense that it would be passing in `2` here as the value of `resourceJson` as it `get`s the value of the first item in the Pair, so it gets the value of `"Priority"` from `consumableResources` in the submit block. So you could refactor this to treat `resourceJson` as an integer, if present.